### DISCUSSION TITLE
Wait and Loop Functions

### DISCUSSION THREAD

**Discussion by EnderdubYT** [4ian/GDevelop#3323] [Enhancement ideas (and their technical discussions)]
So I've recently been working on a Metroidvania, and there have been multiple times that I've thought, "Man, a wait/loop (depending on the situation) function would be really useful here", but I have to find another, less organized way to do it. So I came here to recommend them in hopes of them being added. wait could possibly work something like "Wait [number of seconds]" and then it waits for however long you put to move on to the next function; the loop could have a top and bottom portion that you could put around what you want to loop and put in the parameter of how long or how many you want it to loop for.

**Comment by arthuro555** [comment]
Wait has already an issues and bounty on it: https://github.com/4ian/GDevelop/issues/2975
Repeating X times or while a condition is already possible with their respective event types.

**Reply by EnderdubYT** [reply]
oh, ok, i didnt relize that. Thanks!
